core/src/main/java/org/apache/accumulo/core/client/mapreduce/lib/impl/FileOutputConfigurator.java

```diff
@@ -209,4 +210,12 @@ public static void setSampler(Class<?> implementingClass, Configuration conf, Sa
     }
   }
 
+  public static void setSummarizers(Class<?> implementingClass, Configuration conf, SummarizerConfiguration[] sumarizerConfigs) {
+    Map<String,String> props = SummarizerConfiguration.toTableProperties(sumarizerConfigs);
+
+    for (Entry<String,String> entry : props.entrySet()) {
+      conf.set(enumToConfKey(implementingClass, Opts.ACCUMULO_PROPERTIES) + "." + entry.getKey(), entry.getValue());
```

Comment:
I don't think this code is called frequently enough to justify optimizations that would make the code longer/harder to read.  Also the java compiler will use StringBuilder when concatenating  multiple strings.  However, I don't think it will reuse it across the loop.